homeassistant/components/growatt_server/sensor.py

```diff
@@ -0,0 +1,161 @@
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return "power"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        return "W"
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the monitored installation."""
+        attributes = self.attributes
+        return attributes
+
+    def update(self):
+        """Get the latest data from the Growat API and updates the state."""
+
+        try:
+            self.api.login(self.username, self.password)
+            if self.is_total:
+                total_info = self.api.plant_info(self.inverter_id)
+                del total_info["deviceList"]
+                self._state = total_info["invTodayPpv"]
+                self.attributes = {
+                    "money_today": total_info["plantMoneyText"],
+                    "money_total": total_info["totalMoneyText"],
+                    "energy_today": total_info["todayEnergy"],
+                    "current_wattage": total_info["invTodayPpv"],
+                    "current_voltage": total_info["totalEnergy"],
+                    "maximum_output": total_info["nominalPower"],
+                }
+            else:
+                inverter_info = self.api.inverter_detail(self.inverter_id)
+
+                self._state = inverter_info["data"]["pac"]
+                self.attributes = {
+                    "energy_today": inverter_info["data"]["e_today"],
+                    "energy_total": inverter_info["data"]["e_total"],
+                    "voltage_input_1": inverter_info["data"]["vpv1"],
+                    "amperage_input_1": inverter_info["data"]["ipv1"],
+                    "wattage_input_1": inverter_info["data"]["ppv1"],
+                    "voltage_input_2": inverter_info["data"]["vpv2"],
+                    "amperage_input_2": inverter_info["data"]["ipv2"],
+                    "wattage_input_2": inverter_info["data"]["ppv2"],
+                    "voltage_input_3": inverter_info["data"]["vpv3"],
+                    "amperage_input_3": inverter_info["data"]["ipv3"],
+                    "wattage_input_3": inverter_info["data"]["ppv3"],
+                    "inverter_internal_wattage": inverter_info["data"]["ppv"],
+                    "inverter_reactive_voltage": inverter_info["data"]["vacr"],
+                    "inverter_reactive_amperage": inverter_info["data"]["iacr"],
+                    "frequency": inverter_info["data"]["fac"],
+                    "current_wattage": inverter_info["data"]["pac"],
+                    "current_reactive_wattage": inverter_info["data"]["pacr"],
+                    "fault": inverter_info["data"]["strfault"],
+                    "warning": inverter_info["data"]["strwarning"],
+                    "break": inverter_info["data"]["strbreak"],
```

Comment:
What do fault, warning and break values contain?

Reply:
By default 0.
i (luckily) haven't found out what the value would be otherwise but I'm assuming a string of the warning, fault or break.

Reply:
So it's not a binary value?

Reply:
It could be but i think it will be a string of sorts from the name `STRfault`
![20190825_194549](https://user-images.githubusercontent.com/15870933/63654808-9076f580-c77f-11e9-8deb-25a99eb92701.jpg)
As you can see in this picture the default this api returns is always 0.

And since it's from an api used by their app their documentation is non-existant.

Reply:
Maybe leave those attributes out until we know what they are.